### DISCUSSION TITLE
sidebar/sanfene/mysql

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#86] [Announcements]
# sidebar/sanfene/mysql

图文详解 66 道 MySQL 面试高频题，这次吊打面试官，我觉得稳了（手动 dog）

https://tobebetterjavaer.com/sidebar/sanfene/mysql.html

**Comment by LVXUEYANGTiger** [comment]
已学习

**Comment by xiemeng9462** [comment]
good

**Comment by JesonMrLiu** [comment]
总结得很好，但是有些内容是有问题的，不知道是否有检查好

**Comment by misitetong** [comment]
最后一题SQL有语法错误，可能是因为`rank`是关键字，无法运行，可以改个名字。
```sql
SET @cur_class = NULL, @cur_rank = 0;

SELECT student_name, class, score
FROM (
    SELECT
        student_name,
        class,
        score,
        @cur_rank := IF(@cur_class = class, @cur_rank + 1, 1) AS ranking,
        @cur_class := class
    FROM student_scores
    ORDER BY class, score DESC
) AS ranked
WHERE ranked.ranking <= 10;
```

**Comment by seasurt** [comment]
右连的图搞错了

**Comment by TheFivePercent** [comment]
count(1) = count(*) > count(列名)

**Reply by TheFivePercent** [reply]
这是Oracle官方的说法

**Reply by 0319cc** [reply]
2025.2.19

**Comment by 0319cc** [comment]
2025.2.19

**Comment by gfvmjcdfg** [comment]
这个MVCC的图是不是有点问题啊，是不是应该是新生成的数据的DB_ROLL_PTR指向旧数据啊。

**Comment by DANIELPC59** [comment]
行锁那里，如果加的是普通索引，而非唯一索引，应该还是会出现锁住一大片重复值行的现象把

**Comment by DrLeoChen** [comment]
关于联合索引部分，表述存在问题，“如果创建了一个覆盖索引 idx_users_name_email_age 包含 name、email、age 列：”这里创建的应该是联合索引，覆盖索引应该只是一种SQL优化策略才对，而联合索引才是一种索引类型，才外，后续的索引顺序应该和查询顺序保持一致，否则会违背最左前缀原则，造成索引失效，以上为个人理解，如存在问题请批评指正

**Comment by zslfirst** [comment]
三大范式中第二范式的描述有问题吧，第二范式应该是消除对主键的部分依赖，适用于使用联合主键的场景
